C string error on alias

### Describe the bug

Aliasing a constant C string in a shared library causes error

### Reproduction Steps

```V
pub const text = c'Text'
pub const also_text = &text

pub const text2 = c'Text2'
pub const also_text2 = c'Text2'

pub const text3 = c'Text3'
pub const also_text3 = text3 // Error
```
`v -shared -show-c-output run .`

### Expected Behavior

As a user I would assume to be able to assign an already defined value to another variable.

### Current Behavior

```bash
======== Output of the C Compiler (/home/anton/workspace/v/thirdparty/tcc/tcc.exe) ========
/tmp/v_1000/tmp.01JRXAZAXDX1FWAZ9SXV22GT51.tmp.so.c:12866: error: cannot convert 'struct string' to 'extern unsigned char *'
===========================================================================================
======== Output of the C Compiler (cc) ========
/tmp/v_1000/tmp.01JRXAZAXDX1FWAZ9SXV22GT51.tmp.so.c: In function ‘_vinit’:
/tmp/v_1000/tmp.01JRXAZAXDX1FWAZ9SXV22GT51.tmp.so.c:203:18: error: incompatible types when assigning to type ‘u8 *’ {aka ‘unsigned char *’} from type ‘string’
  203 | #define _SLIT(s) ((string){.str=(byteptr)("" s), .len=(sizeof(s)-1), .is_lit=1})
      |                  ^
/tmp/v_1000/tmp.01JRXAZAXDX1FWAZ9SXV22GT51.tmp.so.c:12866:35: note: in expansion of macro ‘_SLIT’
12866 |         _const_main__also_text3 = _SLIT("Text3");
      |                                   ^~~~~
/tmp/v_1000/tmp.01JRXAZAXDX1FWAZ9SXV22GT51.tmp.so.c:12868:32: warning: assignment to ‘u8 **’ {aka ‘unsigned char **’} from incompatible pointer type ‘char **’ [-Wincompatible-pointer-types]
12868 |         _const_main__also_text = &_const_main__text;
      |                                ^
===============================================
No such file or directory; code: 2
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.10 277c9c5

### Environment details (OS name and version, etc.)

V 0.4.10 277c9c5

--- a/vlib/v/gen/c/consts_and_globals.v
+++ b/vlib/v/gen/c/consts_and_globals.v
@@ -227,10 +227,15 @@ fn (mut g Gen) const_decl_precomputed(mod string, name string, field_name string
 			// TODO: ^ the above for strings, cause:
 			// `error C2099: initializer is not a constant` errors in MSVC,
 			// so fall back to the delayed initialisation scheme:
+			init := if typ == ast.string_type {
+				'_SLIT("${escaped_val}")'
+			} else {
+				'(${styp})"${escaped_val}"'
+			}
 			g.global_const_defs[util.no_dots(field_name)] = GlobalConstDef{
 				mod:   mod
 				def:   '${styp} ${cname}; // str inited later'
-				init:  '\t${cname} = _SLIT("${escaped_val}");'
+				init:  '\t${cname} = ${init};'
 				order: -1
 			}
 			if g.is_autofree {

--- a/vlib/v/tests/consts/const_cstr_test.v
+++ b/vlib/v/tests/consts/const_cstr_test.v
@@ -1,0 +1,14 @@
+pub const global_text = c'Text'
+pub const global_ref = &global_text
+pub const global_copy = global_text
+
+fn test_main() {
+	local_copy := global_text
+	println('${global_text}')
+	println('${global_ref}')
+	println('${local_copy}')
+	println('${global_copy}')
+	assert *global_copy == c'Text'
+	assert **global_ref == c'Text'
+	assert *global_text == c'Text'
+}

